api/src/main/java/org/apache/iceberg/Table.java

```diff
@@ -64,6 +64,17 @@ default IncrementalAppendScan newIncrementalAppendScan() {
     throw new UnsupportedOperationException("Incremental append scan is not supported");
   }
 
+  /**
+   * Create a new {@link IncrementalChangelogScan} for this table.
+   * <p>
+   * Once a scan is created, it can be refined to project columns and filter data.
+   *
+   * @return an incremental changelog scan
+   */
+  default IncrementalChangelogScan newIncrementalChangelogScan() {
```

Comment:
Do we want `changelog` or simply `change`?

Reply:
@stevenzwu, how do you feel about the names? You know more about this CDC use cases. I think I'd slightly prefer shorter names like `ChangeScan` and `ChangeScanTask` but if `changelog` is really widely used, I have no problem using it as well.

Reply:
Seems `changelog` is pretty popular. Let's keep it as-is.